cms/lib/xblock/authoring_mixin.py

```diff
@@ -51,3 +56,111 @@ def visibility_view(self, _context=None):
         scope=Scope.settings,
         enforce_type=True,
     )
+
+    def editor_saved(self, user, old_metadata, old_content) -> None:  # pylint: disable=unused-argument
+        """
+        Called right *before* the block is written to the DB. Can be used, e.g., to modify fields before saving.
+
+        By default, is a no-op. Can be overriden in subclasses.
+        """
+
+    def post_editor_saved(self, user, old_metadata, old_content) -> None:  # pylint: disable=unused-argument
+        """
+        Called right *after* the block is written to the DB. Can be used, e.g., to spin up followup tasks.
+
+        By default, is a no-op. Can be overriden in subclasses.
+        """
+
+    def studio_duplicate(
```

Comment:
^ Note: This is more a question than a request for changes. And probably @kdmccormick and @Agrendalath are the ones best positioned to answer :)

Reply:
@bradenmacdonald,
> I guess my assumption has been that methods on XBlocks or their mixins generally only interact with the core XBlock API, the runtime, and/or runtime services. Any code that's specific to modulestore should be abstracted behind a runtime method or a service. I know we haven't always followed this pattern consistently, but I thought we had been moving in that direction.

Yeah, that makes sense to me. This PR is part of a bigger GitHub issue (https://github.com/openedx/edx-platform/issues/33640), which I haven't been tracking closely enough to answer this, though. @kdmccormick definitely has a better idea about the planned follow-ups.

Reply:
Good point @bradenmacdonald . This does seem like runtime capability rather than a block capability. Until recently I didn't really grasp the distinction between those two :P

@DanielVZ96 and @Agrendalath , when I wrote up the [issue for this refactoring](https://github.com/openedx/edx-platform/issues/33715), I thought it'd be pretty straightfoward and self-contained. In reality, it seems like it's wrapped up in the [complexities of Content Libraries V2](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3966795804/Fun+with+LibraryContentBlock+export+import+and+duplication) and [copy-paste](https://github.com/openedx/edx-platform/pull/33926). I should have warned you folks of that when you originally opened your draft PR; that's my bad. I need to spend the next couple of work days sorting out our plan for duplication, copy-paste, and import for library_content blocks, and then I'll come back with a recommendation of what to do here. Thanks for your responsiveness and patience so far.